File: mvn-advent-of-code/src/test/java/advent/twenty_sixteen/Day04SecurityThroughObscurityTest.java
package advent.twenty_sixteen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day04SecurityThroughObscurityTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day04SecurityThroughObscurity("/twenty_sixteen/Day04-SecurityThroughObscurity-input.txt");

        assertEquals(245102, dp.getPart1Answer().intValue());
        assertEquals(324, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day04SecurityThroughObscurity("/twenty_sixteen/Day04-SecurityThroughObscurity-example#1.txt");

        assertEquals(1514, dp.getPart1Answer().intValue());
        assertEquals(-1, dp.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_sixteen/Day04SecurityThroughObscurityTest.scala
package advent.twenty_sixteen

import org.scalatest.{FlatSpec, Matchers}

class Day04SecurityThroughObscurityTest  extends FlatSpec with Matchers {
  "2016 Day 04 - Input" should "calculate answers" in {
    val dp = new Day04SecurityThroughObscurity("/twenty_sixteen/Day04-SecurityThroughObscurity-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day04SecurityThroughObscurity("/twenty_sixteen/Day04-SecurityThroughObscurity-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
